include/tbb/machine/linux_common.h

```diff
@@ -31,18 +31,39 @@
 
 #if defined(SYS_futex)
 
+/*
+ * Note that this header is included not only on Linux. Some BSD systems may also support futexes,
+ * but provide the definitions in other headers.
+ */
+#if defined(__linux__)
+#include <linux/futex.h>
+#define __TBB_USE_FUTEX 1
+#elif defined(__OpenBSD__)
+#include <sys/futex.h>
 #define __TBB_USE_FUTEX 1
+#elif defined(__has_include)
+#if __has_include(<sys/futex.h>)
+#include <sys/futex.h>
+#define __TBB_USE_FUTEX 1
+#endif
+#endif
+
+#if __TBB_USE_FUTEX
```

Comment:
I suppose the logic with `__TBB_USE_FUTEX` is surplus because all this code is guarded with the [`SYS_futex`](https://github.com/01org/tbb/pull/58/files#diff-1ddd140ec2d4d0fad745786ee9125f29R32) macro (i.e. the futex syscall is supported). It is Ok if no one header file is included because the values of `__TBB_FUTEX_WAIT` and `__TBB_FUTEX_WAKE` are hardcoded to [`0`](https://github.com/01org/tbb/pull/58/files#diff-1ddd140ec2d4d0fad745786ee9125f29R61) and [`1`](https://github.com/01org/tbb/pull/58/files#diff-1ddd140ec2d4d0fad745786ee9125f29R69) respectively in that case.

Reply:
I actually don't like that the macros are defined to some specific values in TBB because the values are obviously system-dependent. Including the header provides these system values.

Reply:
I agree that using headers is the best approach and your patch is worth contributing. However, if TBB is used on the system where the introduced logic will not find the required header file, disabling hardcoded behavior will break backward compatibility.
My suggestion is to include header files where possible (where we know what to include) and not to disable the old behavior for the sake of backward compatibility.

Reply:
Ok, will do.